core/payment/src/models/agreement.rs

```diff
@@ -25,7 +25,7 @@ pub struct WriteObj {
 }
 
 impl WriteObj {
-    pub fn new(agreement: Agreement, role: Role) -> Self {
+    pub fn new(agreement: Agreement, role: Role) -> Result<Self, DbError> {
```

Comment:
```suggestion
    pub fn new(agreement: Agreement, role: Role) -> Result<Self, DbError> {
```
```suggestion
    pub fn try_new(agreement: Agreement, role: Role) -> Result<Self, String> {
```

Reply:
`new` returning an `Result` should always be `try_new`.
It is weird to return `DbError` from constructor. Maybe just return `String`, and then `map_err` it to `DbError` inside the caller of constructor.